ch_lib/civitai.py

```diff
@@ -725,3 +752,65 @@ def check_models_new_version_by_model_types(model_types:list, delay:float=0.2) -
                 new_version_ids.append(version_id)
 
     return new_versions
+
+
+def move_model_to_subfolder(filepath, model_info):
+    modelid = model_info["modelId"]
+
+    util.printD(f"Model ID is {modelid}")
+
+    if modelid == "":
+        return None
+
+    content = civitai_get(f'{URLS["modelId"]}{modelid}')
+
+    util.printD(content)
+
+    tags = content["tags"]
+
+    util.printD(tags)
+
+    # iterate through tags until we find one that matches MODEL_CATEGORIES
+
+    for tag in tags:
+        if tag in MODEL_CATEGORIES:
+            model_category = tag
+
+            # create subfolder if it doesn't exist
+            util.printD("Current filepath is " + filepath)
+
+            # check to make sure the model is not already in the correct subfolder
+            if model_category in filepath:
+                util.printD("Model is already in the correct subfolder")
+                return filepath
+
+            # get the file path without the filename
+            folderpath = os.path.dirname(filepath)
+
+            util.printD("Current folderpath is " + folderpath)
+
+            # create the new folder path
+            new_folderpath = os.path.join(folderpath, model_category)
+
+            util.printD("New folderpath is " + new_folderpath)
+
+            # create the new folder if it doesn't exist
+            if not os.path.exists(new_folderpath):
+                os.makedirs(new_folderpath)
+
+            # move the file to the new folder
+            new_filepath = os.path.join(new_folderpath, os.path.basename(filepath))
+
+            util.printD("New filepath is " + new_filepath)
```

Comment:
perhaps instead of multiple extra lines of logging, it would be better to have one that said something like `f"Moving {old_filepath} to {new_filepath}."`